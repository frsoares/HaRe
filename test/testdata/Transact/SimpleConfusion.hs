module Transact.SimpleConfusion where

import Control.Concurrent

main :: IO ()
main = do
    v1 <- newEmptyMVar
    v2 <- newEmptyMVar

    -- ...

    let v3 = id v2

    -- ...

    let v4 = id v1

    -- ...

    takeMVar v1 >>= putStrLn

