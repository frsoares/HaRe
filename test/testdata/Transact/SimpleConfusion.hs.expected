module Transact.SimpleConfusion where

import Control.Concurrent
import Control.Concurrent.STM

main :: IO ()
main = do
    v1 <- newEmptyMVar
    v2 <- newEmptyTMVarIO

    -- ...

    let v3 = id v2

    -- ...

    let v4 = id v1

    -- ...

    takeMVar v1 >>= putStrLn

